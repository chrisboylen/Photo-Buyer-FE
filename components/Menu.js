import React, { Component } from "react";
import Link from "next/link";
import { MobileStyles } from "./styles/NavStyles";
import User from "../components/User";
import Signout from "./Signout";
import { Mutation } from "react-apollo";
import { TOGGLE_CART_MUTATION } from "./Cart";
import CartCount from "./CartCount";
import PropTypes from "prop-types";

class Menu extends Component {
  render() {
    return (
      <User>
        {({ data: { me } }) => (
          <MobileStyles data-test="mobile">
            <Link href="/items">
              <a data-test="items" onClick={this.props.toggleMenu}>
                Shop
              </a>
            </Link>
            {me && (
              <>
                <Link href="/sell">
                  <a onClick={this.props.toggleMenu}>Sell</a>
                </Link>
                <Link href="/orders">
                  <a onClick={this.props.toggleMenu}>Orders</a>
                </Link>
                <Signout />
                <Mutation mutation={TOGGLE_CART_MUTATION}>
                  {toggleCart => (
                    <button onClick={toggleCart} data-test="signout">
                      My Cart
                      <CartCount
                        count={me.cart.reduce(
                          (tally, cartItem) => tally + cartItem.quantity,
                          0
                        )}
                      />
                    </button>
                  )}
                </Mutation>
              </>
            )}
            {!me && (
              <>
                <Link href="/signup">
                  <a onClick={this.props.toggleMenu}>Signup</a>
                </Link>
                <Link href="/signin">
                  <a onClick={this.props.toggleMenu}>Signin</a>
                </Link>
              </>
            )}
          </MobileStyles>
        )}
      </User>
    );
  }
}

Menu.propTypes = {};

export default Menu;
