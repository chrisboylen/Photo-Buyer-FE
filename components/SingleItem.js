import React, { Component } from "react";
import gql from "graphql-tag";
import { Query } from "react-apollo";
import Error from "./ErrorMessage";
import styled from "styled-components";
import Head from "next/head";
import formatMoney from "../lib/formatMoney";

const SingleItemStyles = styled.div`
  border-radius: 5px;
  max-width: 1200px;
  margin: 1rem auto;
  box-shadow: ${props => props.theme.bs};
  display: grid;
  grid-auto-columns: 1fr;
  grid-auto-flow: column;
  min-height: 800px;
  h2 {
    background: ${props => props.theme.blue};
    border-radius: 5px 5px 0 0;
    color: ${props => props.theme.white};
    text-align: center;
    margin: 0;
    span {
      color: #c7e4ff;
    }
  }
  img {
    width: 100%;
    height: 100%;
    object-fit: contain;
    border-radius: 0 0 5px 5px;
  }
  .details {
    margin: 3rem;
    font-size: 2rem;
  }
`;

const SINGLE_ITEM_QUERY = gql`
  query SINGLE_ITEM_QUERY($id: ID!) {
    item(where: { id: $id }) {
      id
      title
      description
      largeImage
    }
  }
`;
class SingleItem extends Component {
  render() {
    return (
      <Query
        query={SINGLE_ITEM_QUERY}
        variables={{
          id: this.props.id
        }}
      >
        {({ error, loading, data }) => {
          if (error) return <Error error={error} />;
          if (loading) return <p>Loading...</p>;
          if (!data.item) return <p>No Item Found for {this.props.id}</p>;
          const item = data.item;
          return (
            <SingleItemStyles>
              <Head>
                <title>Photo Buyer | {item.title}</title>
              </Head>
              <div className="details">
                <h2>
                  <span>Viewing: </span> {item.title}
                </h2>
                <img src={item.largeImage} alt={item.title} />
                <p>{item.description}</p>
              </div>
            </SingleItemStyles>
          );
        }}
      </Query>
    );
  }
}

export default SingleItem;
export { SINGLE_ITEM_QUERY };
