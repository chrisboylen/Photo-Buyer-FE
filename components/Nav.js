import React, { Component } from "react";
import Link from "next/link";
import { NavStyles, MobileStyles } from "./styles/NavStyles";
import User from "./User";
import Signout from "./Signout";
import { Mutation } from "react-apollo";
import { TOGGLE_CART_MUTATION } from "./Cart";
import CartCount from "./CartCount";
import Menu from "./Menu";

class Nav extends Component {
  state = {
    isOpen: false
  };

  toggleMenu = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  render() {
    return (
      <User>
        {({ data: { me } }) => (
          <NavStyles data-test="nav">
            {this.state.isOpen && (
              <Menu isOpen={this.state.isOpen} toggleMenu={this.toggleMenu} />
            )}
            <button
              onClick={this.toggleMenu}
              className="mobile-checkbox"
              id="mobile-toggle"
            />
            <label htmlFor="mobile-toggle" className="mobile-btn">
              <span
                className={
                  this.state.isOpen
                    ? "mobile-icon mobile-icon-open"
                    : "mobile-icon"
                }
              >
                &nbsp;
              </span>
            </label>
            <div className="mobile-bkg">&nbsp;</div>
            <div className="web-nav">
              <Link href="/items">
                <a>Shop</a>
              </Link>
              {me && (
                <>
                  <Link href="/sell">
                    <a>Sell</a>
                  </Link>
                  <Link href="/orders">
                    <a>Orders</a>
                  </Link>
                  <Signout />
                  <Mutation mutation={TOGGLE_CART_MUTATION}>
                    {toggleCart => (
                      <button onClick={toggleCart}>
                        My Cart
                        <CartCount
                          count={me.cart.reduce(
                            (tally, cartItem) => tally + cartItem.quantity,
                            0
                          )}
                        />
                      </button>
                    )}
                  </Mutation>
                </>
              )}
              {!me && (
                <>
                  <Link href="/signup">
                    <a>Signup</a>
                  </Link>
                  <Link href="/signin">
                    <a>Signin</a>
                  </Link>
                </>
              )}
            </div>
          </NavStyles>
        )}
      </User>
    );
  }
}

export default Nav;
