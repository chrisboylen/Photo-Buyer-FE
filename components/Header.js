import Link from "next/link";
import styled from "styled-components";
import Nav from "./Nav";
import Router from "next/router";
import NProgress from "nprogress";
import Cart from "./Cart";
import Search from "./Search";

Router.onRouteChangeStart = () => {
  NProgress.start();
};
Router.onRouteChangeComplete = () => {
  NProgress.done();
};
Router.onRouteChangeError = () => {
  NProgress.done();
};

const Logo = styled.h1`
  background: ${props => props.theme.lightblue};
  font-size: 8rem;
  margin: 0;
  min-height: 12rem;
  position: relative;
  z-index: 1500;
  a {
    margin: 0 0 0 10rem;
    padding: 1rem 2rem 0.6rem 2rem;
    border-radius: 8px;
    background: #e8f4ff;
    color: #4dadfe;
    text-transform: uppercase;
    text-decoration: none;
  }
  span {
    background: linear-gradient(to right, #00f1fe 0%, #4dadfe 100%);
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
  img {
    width: 6rem;
    margin: 0 2rem 0 0;
  }
  @media (max-width: 1300px) {
    font-size: 6rem;
    img {
      width: 4.5rem;
    }
    @media (max-width: 760px) {
      line-height: 1.5;
      a {
        top: 2rem;
        left: 3rem;
        margin: 0;
        padding: 0.3rem 1.1rem 0 1.1rem;
        position: relative;
      }
      img {
        margin: 0;
        width: 4rem;
      }
      span {
        display: none;
      }
    }
  }
`;

const StyledHeader = styled.header`
  left: 0;
  top: 0;
  position: sticky;
  z-index: 500;
  .bar {
    grid-template-columns: auto 1fr;
    justify-content: space-between;
    align-items: stretch;
    @media (max-width: 1300px) {
      grid-template-columns: 1fr;
      justify-content: center;
    }
  }
  .sub-bar {
    background: linear-gradient(
      to bottom,
      #8cbce2 0%,
      #8cbce2 70%,
      #193e5d 100%
    );
    display: grid;
    grid-template-columns: 1fr;
    border-bottom: 1px solid ${props => props.theme.lightgrey};
  }
`;

const Header = () => (
  <StyledHeader>
    <div className="bar">
      <Logo>
        <Link href="/">
          <a>
            <img src="../static/camera.png" alt="Camera Icon" />
            <span className="text">Photo Buyer</span>
          </a>
        </Link>
      </Logo>
      <Nav />
    </div>
    <div className="sub-bar">
      <Search />
    </div>
    <Cart />
  </StyledHeader>
);

export default Header;
