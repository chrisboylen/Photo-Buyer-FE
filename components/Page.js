import React, { Component } from "react";
import styled, { ThemeProvider, injectGlobal } from "styled-components";
import Header from "./Header";
import Meta from "./Meta";

const theme = {
  lightblue: "#9ed2ff",
  blue: "#4dadfe",
  black: "#193e5d",
  grey: "#3a3a3a",
  lightgrey: "#b1c0e0",
  white: "#e8f4ff",
  offWhite: "#ededed",
  maxWidth: "1000px",
  bs: "0 12px 24px 0 rgba(0, 0, 0, 0.21)"
};

const StyledPage = styled.div`
  color: ${props => props.theme.black};
`;

const Inner = styled.div`
  max-width: ${props => props.theme.maxWidth};
  margin: 0 auto;
  padding: 2rem;
  @media (max-width: 760px) {
    padding: 0;
  }
`;

injectGlobal`
  @font-face {
    font-family: 'DoHyeon-Regular';
    src: url('/static/DoHyeon-Regular.ttf')
    format('woff2');
    font-weight: normal;
    font-style: normal;
  }
  html {
    box-sizing: border-box;
    font-size: 10px;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  body {
    background: #e8f4ff52;
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height: 2;
    font-family: 'DoHyeon-Regular';
  }
  a {
    text-decoration: none;
    color: ${theme.black}
  }
`;

class Page extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <StyledPage>
          <Meta />
          <Header />
          <Inner>{this.props.children}</Inner>
        </StyledPage>
      </ThemeProvider>
    );
  }
}

export default Page;
