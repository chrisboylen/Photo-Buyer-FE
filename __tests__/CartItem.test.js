import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";
import { fakeUser, fakeCartItem } from "../lib/testUtils";
import { CURRENT_USER_QUERY } from "../components/User";
import CartItem from "../components/CartItem";
import RemoveFromCart from "../components/RemoveFromCart";

const mocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: {
      data: {
        me: {
          ...fakeUser(),
          cart: [fakeCartItem()]
        }
      }
    }
  }
];
const cartItem = {
  id: "5",
  item: {
    id: 1,
    title: "landscape"
  }
};

describe("CART ITEM", () => {
  it("renders and matches snapshot with cart item", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <CartItem cartItem={cartItem} />
        <RemoveFromCart id={cartItem.id} />
      </MockedProvider>
    );

    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find("CartItem"))).toMatchSnapshot();
  });

  it("renders and matches snapshot without cart item", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <CartItem cartItem={{}} />
        <RemoveFromCart id={cartItem.id} />
      </MockedProvider>
    );

    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find("CartItem"))).toMatchSnapshot();
  });
});
