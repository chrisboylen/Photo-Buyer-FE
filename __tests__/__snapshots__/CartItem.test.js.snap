// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CART ITEM renders and matches snapshot with cart item 1`] = `
<CartItem
  cartItem={
    Object {
      "id": "5",
      "item": Object {
        "id": 1,
        "title": "landscape",
      },
    }
  }
>
  <CartItem__CartItemStyles>
    <li
      className="CartItem__CartItemStyles-xrdx0c-0 fQTlwp"
    >
      <img
        alt="landscape"
        width="100"
      />
      <div
        className="cart-item-details"
      >
        <h3>
          landscape
        </h3>
        <p>
          $NaN
          -
          <em>
             × 
            $NaN
             each
          </em>
        </p>
      </div>
      <RemoveFromCart
        id="5"
      >
        <Mutation
          mutation={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "removeFromCart",
                  },
                  "operation": "mutation",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "removeFromCart",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [
                    Object {
                      "defaultValue": undefined,
                      "kind": "VariableDefinition",
                      "type": Object {
                        "kind": "NonNullType",
                        "type": Object {
                          "kind": "NamedType",
                          "name": Object {
                            "kind": "Name",
                            "value": "ID",
                          },
                        },
                      },
                      "variable": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                      },
                    },
                  ],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 88,
                "start": 0,
              },
            }
          }
          optimisticResponse={
            Object {
              "__typename": "Mutation",
              "removeFromCart": Object {
                "__typename": "CartItem",
                "id": "5",
              },
            }
          }
          update={[Function]}
          variables={
            Object {
              "id": "5",
            }
          }
        >
          <RemoveFromCart__BigButton
            disabled={false}
            onClick={[Function]}
            title="Delete Item"
          >
            <button
              className="RemoveFromCart__BigButton-sc-1tq5xu5-0 WgPXB"
              disabled={false}
              onClick={[Function]}
              title="Delete Item"
            >
              ×
            </button>
          </RemoveFromCart__BigButton>
        </Mutation>
      </RemoveFromCart>
    </li>
  </CartItem__CartItemStyles>
</CartItem>
`;

exports[`CART ITEM renders and matches snapshot without cart item 1`] = `
<CartItem
  cartItem={Object {}}
>
  <CartItem__CartItemStyles>
    <li
      className="CartItem__CartItemStyles-xrdx0c-0 fQTlwp"
    >
      <p>
        This item has been removed!
      </p>
    </li>
  </CartItem__CartItemStyles>
</CartItem>
`;
